FreeRTOS/FreeRTOS-Kernel
# Fix/clear mie bit in initial risc v mstatus

fix: CLEAR MIE BIT IN INITIAL RISC-V MSTATUS VALUE

Description
-----------
The MIE bit in the RISC-V MSTATUS register is used to globally enable
or disable interrupts.  It is copied into the MPIE bit and cleared
on entry to an interrupt, and then copied back from the MPIE bit on
exit from an interrupt.

When a task is created it is given an initial MSTATUS value that is
derived from the current MSTATUS value with the MPIE bit force to 1,
but the MIE bit is not forced into any state.  This change forces
the MIE bit to 0 (interrupts disabled).

Test Steps
-----------
Run the comprehensive demo for the board.

Related Issue
-----------
https://forums.freertos.org/t/risc-v-port-pxportinitialisestack-issue-about-mstatus-value-onto-the-stack/9622


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
